src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlan.java

```diff
@@ -97,21 +94,34 @@ public void execute() throws ValidationEngineException {
 					agpCheck.getAGPEntries();
 				}
 			}
-
-			if(options.context.get().getFileTypes().contains(FileType.FASTA))
-			  validateFasta();
+			if(options.context.get().getFileTypes().contains(FileType.FASTA)) {
+				validationResult = validateFasta();
+				if(!validationResult.isValid())
+					return validationResult;
+			}
```

Comment:
`	for(SubmissionFile fastaFile:options.submissionFiles.get().getFiles(FileType.FASTA))
			{
				fileName= fastaFile.getFile().getName();
				if(sequenceDB!=null)
					check.setSequenceDB(sequenceDB);
				if(contigDB!=null)
					check.setContigDB(contigDB);
				if(!check.check(fastaFile))
					throwValidationCheckException(FileType.FASTA,fastaFile);
				else if(!options.isRemote)
				     flagValidation(FileType.FASTA);

			}`

Nothing changed here ,except we return validationResult instead of throwing exception. And I moved flagfile creation part alone out of for loop, bcoz it makes sense create that after all files are validated

Reply:
ok